Source/Core/Gen8/Generators/WildGeneratorPLA.cpp

```diff
@@ -0,0 +1,142 @@
+#include "WildGeneratorPLA.hpp"
+#include <Core/Enum/Method.hpp>
+#include <Core/Parents/States/State.hpp>
+#include <Core/RNG/RNGList.hpp>
+#include <Core/RNG/Xoroshiro.hpp>
+
+WildGeneratorPLA::WildGeneratorPLA(u32 initialAdvances, u32 maxAdvances, u16 tid, u16 sid, u8 genderRatio, const StateFilter &filter,
+                                   u8 rolls, bool isAlpha) :
+    Generator(initialAdvances, maxAdvances, tid, sid, genderRatio, Method::None, filter), rolls(rolls), isAlpha(isAlpha)
+{
+}
+
+std::vector<State> WildGeneratorPLA::generate(u64 seed) const
+{
+    std::vector<State> states;
+
+    seed -= 0x82A2B175229D6A5B;
+
+    for (u32 advance = 0; advance < initialAdvances; advance++)
+    {
+        // Probably horrible implementation but I forget how RNGList works
+        Xoroshiro mainRng(seed);
+        mainRng.next();
+        mainRng.next();
+        seed = mainRng.next();
+    }
+
+    for (u32 advance = 0; advance <= maxAdvances; advance++)
+    {
+        State result(initialAdvances + advance);
+        Xoroshiro mainRng(seed);
```

Comment:
Would making use of RNGList be possible here. We are using several different Xoroshiros here

Reply:
Honestly I didn't even think of using it because of all the Xoroshiro instances. Don't believe it would be easily possible as it needs to reseed the whole xoroshiro of the first to then create the rest again